src/tasks/buildArtifact/BuildArtifactTaskProvider.ts

```diff
@@ -99,30 +99,37 @@ export class BuildArtifactTaskProvider implements TaskProvider {
     }
 
     public static async resolveExportTask(task: Task, type: string): Promise<Task> {
-        if (!await languageServerApiManager.ready()) {
+        const languageServerStatus = await Promise.race([languageServerApiManager.ready(), BuildArtifactTaskProvider.getLanguageServerStatusTimeout()]);
+        if (!languageServerStatus) {
             return task;
         }
         const definition: IExportJarTaskDefinition = <IExportJarTaskDefinition>task.definition;
         const folder: WorkspaceFolder = <WorkspaceFolder>task.scope;
-        const resolvedTask: Task = new Task(definition, folder, task.name, type,
-            new CustomExecution(async (resolvedDefinition: IExportJarTaskDefinition): Promise<Pseudoterminal> => {
-                const stepMetadata: IStepMetadata = {
-                    entry: undefined,
-                    taskLabel: resolvedDefinition.label || folder.name,
-                    workspaceFolder: folder,
-                    projectList: await Jdtls.getProjects(folder.uri.toString()),
-                    steps: [],
-                    elements: [],
-                    classpaths: [],
-                };
-                return new ExportJarTaskTerminal(resolvedDefinition, stepMetadata);
-            }));
-        resolvedTask.presentationOptions.reveal = TaskRevealKind.Never;
-        return resolvedTask;
+        try {
+            const projectList = await Jdtls.getProjects(folder.uri.toString());
+            const resolvedTask: Task = new Task(definition, folder, task.name, type,
+                new CustomExecution(async (resolvedDefinition: IExportJarTaskDefinition): Promise<Pseudoterminal> => {
+                    const stepMetadata: IStepMetadata = {
+                        entry: undefined,
+                        taskLabel: resolvedDefinition.label || folder.name,
+                        workspaceFolder: folder,
+                        projectList: projectList,
+                        steps: [],
+                        elements: [],
+                        classpaths: [],
+                    };
+                    return new ExportJarTaskTerminal(resolvedDefinition, stepMetadata);
+                }));
+            resolvedTask.presentationOptions.reveal = TaskRevealKind.Never;
+            return resolvedTask;
+        } catch (e) {
+            return task;
+        }
     }
 
     public async provideTasks(): Promise<Task[] | undefined> {
-        if (!await languageServerApiManager.ready()) {
+        const languageServerStatus = await Promise.race([languageServerApiManager.ready(), BuildArtifactTaskProvider.getLanguageServerStatusTimeout()]);
```

Comment:
And let's extract this Promise.race statement to a method with document explaining the reason doing this.

Reply:
in https://github.com/microsoft/vscode-java-dependency/pull/722/commits/0f4564ca738ba542897b646f97219d1f8cc32608 we:

1. extract the utility to languageServerApiManager, named `isReady(timeout)`
2. rename original `isReady` field to `isServerReady`